src/test/java/io/github/thebusybiscuit/slimefun4/testing/tests/utils/TestInfiniteBlockGenerators.java

```diff
@@ -0,0 +1,162 @@
+package io.github.thebusybiscuit.slimefun4.testing.tests.utils;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.event.block.BlockFromToEvent;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import be.seeseemelk.mockbukkit.MockBukkit;
+import be.seeseemelk.mockbukkit.ServerMock;
+import be.seeseemelk.mockbukkit.WorldMock;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.utils.InfiniteBlockGenerator;
+
+class TestInfiniteBlockGenerators {
+
+    private static ServerMock server;
+
+    @BeforeAll
+    public static void load() {
+        server = MockBukkit.mock();
+        MockBukkit.load(SlimefunPlugin.class);
+    }
+
+    @AfterAll
+    public static void unload() {
+        MockBukkit.unmock();
+    }
+
+    @Test
+    @DisplayName("Test if invalid Cobblestone generators are ignored")
+    void testInvalidCobblestoneGenerator() {
+        World world = new WorldMock();
+        Block block = world.getBlockAt(0, 100, 0);
+
+        block.setType(Material.STONE);
+        Assertions.assertFalse(InfiniteBlockGenerator.COBBLESTONE_GENERATOR.test(block));
+        Assertions.assertNull(InfiniteBlockGenerator.findAt(block, false));
+
+        block.setType(Material.COBBLESTONE);
+        Assertions.assertFalse(InfiniteBlockGenerator.COBBLESTONE_GENERATOR.test(block));
+        Assertions.assertNull(InfiniteBlockGenerator.findAt(block, false));
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "provideFaces")
+    @DisplayName("Test if a Cobblestone Generator can be detected")
+    void testValidCobblestoneGenerator(BlockFace water, BlockFace lava) {
```

Comment:
Can't seem to link to this line so I'm just commenting, these two arguments. I'm not sure how Unit Tests handle args but if water and lava refer to the same BlockFace you get no ValidCobblestoneGenerator

Reply:
There is a not-equals check for the arguments. Look at the annotations, they point to "provideFaces", a method found further down. There is a .filter() for this exact case.